i18n/src/main/scala/io/udash/i18n/TranslationKey.scala

```diff
@@ -97,12 +100,23 @@ object TranslationKey {
     new Untranslatable(key)
 
   private[i18n] final class ReducedTranslationKey(key: String, val argv: Any*) extends TranslationKey0(key) {
+    override def productPrefix: String = "ReducedTranslationKey"
+    override def productArity: Int = 2
+    override def productElement(n: Int): Any = n match {
```

Comment:
no real preference yet, but have you considered:
```scala
    override def productArity: Int = argv.size + 1
    override def productElement(n: Int): Any = if(n == 0) key else argv(n - 1)
```

One advantage is you wouldn't have to override `toString` :P

Reply:
I did it this way because:

1. I felt that `Product` implementations have all "hardoded" product arity, so that one class definition can have only one arity.
2. I wanted to mimic adding `case`exactly - but then made exception with `toString` as `"ReducedTranslationKey(key1,WrappedArray(1, 2, 3))"` was just too ugly. :p

Both arguments are rather weak, so no clear preference on my part.